src/cloud/components/Views/Table/TableViewContentManager.tsx

```diff
@@ -443,26 +506,54 @@ const TableViewContentManager = ({
           {orderedDocs.length === 0 && <EmptyRow label='No Documents' />}
           {currentWorkspaceId != null && (
             <TableContentManagerRow>
-              <Button
-                className='content__manager--no-padding'
-                variant='transparent'
-                iconPath={mdiPlus}
-                onClick={() =>
-                  openNewDocForm(
-                    {
-                      team,
-                      parentFolderId: currentFolderId,
-                      workspaceId: currentWorkspaceId,
-                    },
-                    {
-                      skipRedirect: true,
-                      precedingRows: [],
+              {newDocRowState === 'idle' ? (
+                <Button
+                  className='content__manager--no-padding'
+                  variant='transparent'
+                  iconPath={mdiPlus}
+                  onClick={() => setNewDocRowState('editing')}
+                >
+                  {translate(lngKeys.ModalsCreateNewDocument)}
+                </Button>
+              ) : newDocRowState === 'editing' ? (
+                <FormInput
+                  ref={newDocInputRef}
+                  value={newDocName}
+                  onChange={(event) => {
+                    setNewDocName(event.target.value)
+                  }}
+                  onCompositionStart={() => {
+                    newDocCompositionStateRef.current = true
+                  }}
+                  onCompositionEnd={() => {
+                    newDocCompositionStateRef.current = false
+                    if (newDocInputRef.current != null) {
+                      newDocInputRef.current.focus()
                     }
-                  )
-                }
-              >
-                {translate(lngKeys.ModalsCreateNewDocument)}
-              </Button>
+                  }}
+                  onKeyPress={(event) => {
+                    if (newDocCompositionStateRef.current) {
+                      return
+                    }
+                    switch (event.key) {
+                      case 'Escape':
+                        event.preventDefault()
+                        setNewDocRowState('idle')
+                        setNewDocName('')
+                        return
+                      case 'Enter':
+                        event.preventDefault()
+                        createNewDoc()
+                        return
+                    }
+                  }}
+                  onBlur={() => {
+                    createNewDoc()
```

Comment:
to be discussed but I would rather have the onBlur being an `escape` option UX wise to make it similar to the sidebar and just keep enter as the only creation confirmation.